### ISSUE TITLE
How to write NavigationEpisode to json.gz file?

### ISSUE THREAD

**Issue by stbook** [facebookresearch/habitat-api#246]
## ❓ Questions and Help
I was trying to build my own training episodes by generating start positions and goal positions and NavigationEpisodes.
But when I was writing the built NavigationEpisode into .json.gz file, I get this error
'TypeError: Object of type 'NavigationEpisode' is not JSON serializable'

I couldn't fix it. I was wondering how the scene_name.json.gz file were created?

**Comment by mathfac** [comment]
@stbook, are you sure you use `dataset.to_json()` method that employs [Dataset's custom JSON Encoder](https://github.com/facebookresearch/habitat-api/blob/master/habitat/core/dataset.py#L113)?

**Comment by stbook** [comment]
Thanks for your quick response. 

I changed `dataset.to_json()` a bit by converting `numpy.float32` to `float` and it works now.

I have a follow-up question. After we created the dataset should we use the following line to write it directly to disk, because I run into some errors here. I was wondering if there is a better way or an API to use.
`with gzip.GzipFile('../data/datasets/pointnav/mp3d/v1/my/temp.json.gz', 'w') as fout:
    fout.write(dataset)`
